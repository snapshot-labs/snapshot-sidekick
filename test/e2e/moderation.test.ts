import request from 'supertest';
import db from '../../src/helpers/mysql';
import { SqlFixtures as moderation } from '../fixtures/moderation';

const HOST = `http://localhost:${process.env.PORT || 3003}`;

describe('GET /api/moderation', () => {
  beforeAll(async () => {
    const data = Object.values(moderation)
      .flat()
      .map(d => Object.values(d));
    await db.queryAsync(`INSERT INTO moderation (action, type, value, created) VALUES ?`, [
      data as any
    ]);
  });

  afterAll(async () => {
    await db.queryAsync('DELETE FROM sidekick_test.moderation;');
    await db.endAsync();
  });

  describe('when list params is empty', () => {
    it('returns all the list', async () => {
      const response = await request(HOST).get('/api/moderation');

      expect(response.statusCode).toBe(200);
      expect(response.body).toMatchSnapshot();
    });
  });

  describe('when list params is set', () => {
    it.each(['flaggedLinks', 'verifiedTokens'])(
      'returns only the selected %s list',
      async field => {
        const response = await request(HOST).get(`/api/moderation?list=${field}`);

        expect(response.statusCode).toBe(200);
        expect(response.body).toMatchSnapshot();
      }
    );
  });

  it('returns multiple list: flaggedLinks,flaggedIps', async () => {
    const response = await request(HOST).get(`/api/moderation?list=flaggedLinks,flaggedIps`);

    expect(response.statusCode).toBe(200);
    expect(response.body).toMatchSnapshot();
  });

  it('ignores invalid field, and returns only flaggedLinks', async () => {
    const response = await request(HOST).get(`/api/moderation?list=flaggedLinks,testInvalid`);

    expect(response.statusCode).toBe(200);
    expect(response.body).toMatchSnapshot();
  });
});
