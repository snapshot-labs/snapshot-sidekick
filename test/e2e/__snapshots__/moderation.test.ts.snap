// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GET /api/moderation ignores invalid field, and returns only flaggedLinks 1`] = `
{
  "flaggedLinks": [
    "https://facebook.com",
    "https://gogle.com",
  ],
}
`;

exports[`GET /api/moderation returns multiple list: flaggedLinks,flaggedIps 1`] = `
{
  "flaggedIps": [
    "12ca17b49af2289436f303e0166030a21e525d266e209267433801a8fd4071a0",
    "19e36255972107d42b8cecb77ef5622e842e8a50778a6ed8dd1ce94732daca9e",
  ],
  "flaggedLinks": [
    "https://facebook.com",
    "https://gogle.com",
  ],
}
`;

exports[`GET /api/moderation when list params is empty returns all the list 1`] = `
{
  "flaggedAddresses": [
    "0x0001",
    "0x0002",
  ],
  "flaggedIps": [
    "12ca17b49af2289436f303e0166030a21e525d266e209267433801a8fd4071a0",
    "19e36255972107d42b8cecb77ef5622e842e8a50778a6ed8dd1ce94732daca9e",
  ],
  "flaggedLinks": [
    "https://facebook.com",
    "https://gogle.com",
  ],
  "verifiedTokens": {
    "test": {
      "value": "a",
    },
  },
}
`;

exports[`GET /api/moderation when list params is set returns only the selected flaggedLinks list 1`] = `
{
  "flaggedLinks": [
    "https://facebook.com",
    "https://gogle.com",
  ],
}
`;

exports[`GET /api/moderation when list params is set returns only the selected verifiedTokens list 1`] = `
{
  "verifiedTokens": {
    "test": {
      "value": "a",
    },
  },
}
`;
